core/store/src/trie/trie_storage.rs

```diff
@@ -90,14 +90,21 @@ struct TrieCacheMetrics {
 }
 
 impl TrieCacheInner {
+    /// Assumed number of bytes used to store an entry in the cache.
+    ///
+    /// 100 bytes is an approximation based on lru 0.7.5.
+    pub(crate) const PER_ENTRY_OVERHEAD: u64 = 100;
+
     pub(crate) fn new(
-        cache_capacity: usize,
         deletions_queue_capacity: usize,
         total_size_limit: u64,
         shard_id: ShardId,
         is_view: bool,
     ) -> Self {
-        assert!(cache_capacity > 0 && total_size_limit > 0);
+        assert!(total_size_limit > 0);
```

Comment:
Could you sanity-check that the behavior is correct when `total_size_limit` is less than even a single entry?

Reply:
I added a test that shows that we can still insert one value. I hope I understood your concern right and that addresses it, otherwise let me know.